client/src/app/payroll/payroll-list/payroll-list.component.ts

```diff
@@ -1,20 +1,38 @@
-import { Component } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
+
 import { faSkull, faSmileBeam } from '@fortawesome/free-solid-svg-icons';
-import { DataService } from 'src/app/data/data-service';
-import { Payroll } from 'src/app/data/state/payroll-model';
+
+import { PayrollListQuery } from './state/payroll-list.query';
+import { PayrollListService } from './state/payroll-list.service';
 
 @Component({
   selector: 'app-payroll-list',
   templateUrl: './payroll-list.component.html',
   styleUrls: ['./payroll-list.component.scss'],
 })
-export class PayrollListComponent {
+export class PayrollListComponent implements OnInit {
   readonly faSkull = faSkull;
   readonly faSmileBeam = faSmileBeam;
+  readonly payrolls = this.query.selectAll();
+  private employeeDepartment: string;
+
+  @Input()
+  set department(department: string) {
+    const employeeDepartment = department && department.trim();
+    this.service.getPayrolls(employeeDepartment);
+    this.employeeDepartment = employeeDepartment;
+  }
+
+  get department(): string {
+    return this.employeeDepartment;
+  }
 
-  constructor(private dataService: DataService) {}
+  constructor(
+    private query: PayrollListQuery,
+    private service: PayrollListService,
+  ) {}
 
-  getPayroll(): Payroll[] {
-    return this.dataService.getPayroll();
+  ngOnInit() {
+    this.service.getPayrolls(this.department);
```

Comment:
The use of an `@Input()` and a `Query` in this component could be an indication of mixed-purposes.

This component could be looking more and more like a presentational component that should accept `@Input()` of employees and not have any service dependencies, which would move the orchestration/data access logic one level up.

Reply:
Refactored to remove `Service` and `Query` and added an input `PayrollListItem[]` from the `PayrollListComponent`. `PayrollComponent` now uses the `Service` and `Query` to get the data and push it into `PayrollListComponent`